### PR TITLE
fix(CommonITILObject): clone actors

### PR DESCRIPTION
**Pull request by stonebuzz**
Actors of a ```CommonITILObject``` object are not cloned

This PR fix this

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Comment by stonebuzz** [comment]
@orthagh is there a reason why the actors are no longer used in a clone?

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -2728,7 +2728,7 @@ public function post_clone($source, $history)
 
     public function getCloneRelations(): array
     {
-        return [];
+        return [$this->userlinkclass, $this->grouplinkclass, $this->supplierlinkclass];
```
Comment (cedric-anne) [inline src/CommonITILObject.php:2731]: Plugins implementations of `CommonITILObject` may not define some of these relations.
```suggestion
        $relations = [];
        
        if (is_a($this->userlinkclass, CommonITILActor::class, true)) {
            $relations[] = $this->userlinkclass;
        }
        if (is_a($this->grouplinkclass, CommonITILActor::class, true)) {
            $relations[] = $this->grouplinkclass;
        }
        if (is_a($this->supplierlinkclass, CommonITILActor::class, true)) {
            $relations[] = $this->supplierlinkclass;
        }
        
        return $relations;
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Comment by orthagh** [comment]
> @orthagh is there a reason why the actors are no longer used in a clone?

Why this question, did I specificaly removed the code ?
I don't remember.

**Comment by stonebuzz** [comment]
> > @orthagh is there a reason why the actors are no longer used in a clone?
> 
> Why this question, did I specificaly removed the code ? I don't remember.

@cedric-anne  thought there might be a good reason for this

**Pull request merged**